src/parser.mjs

```diff
@@ -444,6 +469,131 @@ export class ExifParser extends Reader {
 		}
 	}
 
+	// FLIR file header (ref 3)
+	// 0x00 - 00 - string[4] file format ID = "FFF\0"
+	// 0x04 - 04 - string[16] file creator: seen "\0","MTX IR\0","CAMCTRL\0"
+	// 0x14 - 20 - int32u file format version = 100
+	// 0x18 - 24 - int32u offset to record directory
+	// 0x1c - 28 - int32u number of entries in record directory
+	// 0x20 - 32 - int32u next free index ID = 2
+	// 0x24 - 36 - int16u swap pattern = 0 (?)
+	// 0x28 - 40 - int16u[7] spares
+	// 0x34 - 52 - int32u[2] reserved
+	// 0x3c - 60 - int32u checksum
+	parseFlirFFFSegment() {
+		// Cancel if the file doesn't contain the segment or if it's damaged.
+		if (!this.ensureSegmentPosition('flir', findFlirFFF)) return
+
+		let le = this.le
+		let flir = {}
+
+		const creator = toString(this.buffer, this.flirOffset + 4, this.flirOffset + 4 + 16, true)
+		flir.creator = creator
+
+		let version = getUint32(this.buffer, this.flirOffset + 20, le)
+		// The version should lie in the 100's. If it doesn't we're reading with the wrong byte order.
+		if (version < 100 || version >= 200) {
+			le = !le
+			version = getUint32(this.buffer, this.flirOffset + 20, le)
+		}
+		flir.version = version
+
+		this.parseFlirFFFDirectory(flir, le)
+		this.flir = flir
+	}
+
+    parseFlirFFFDirectory(flir, le) {
+		let directoryOffset = getUint32(this.buffer, this.flirOffset + 24, le)
+		directoryOffset += this.flirOffset
+
+		const numberOfEntries = getUint32(this.buffer, this.flirOffset + 28, le)
+
+		// Parse table
+		// Every record is 32 bytes
+		for (let i = 0; i < numberOfEntries; i += 1) {
+			const recordOffset = directoryOffset + i * 32
+			this.parseFlirFFFRecord(flir, recordOffset, le)
+		}
+	}
+
+	// FLIR record entry (ref 3):
+	// 0x00 - 00 - int16u record type
+	// 0x02 - 02 - int16u record subtype: RawData 1=BE, 2=LE, 3=PNG; 1 for other record types
+	// 0x04 - 04 - int32u record version: seen 0x64,0x66,0x67,0x68,0x6f,0x104
+	// 0x08 - 08 - int32u index id = 1
+	// 0x0c - 12 - int32u record offset from start of FLIR data
+	// 0x10 - 16 - int32u record length
+	// 0x14 - 20 - int32u parent = 0 (?)
+	// 0x18 - 24 - int32u object number = 0 (?)
+	// 0x1c - 28 - int32u checksum: 0 for no checksum
+	parseFlirFFFRecord(flir, recordOffset, le) {
+		const recordType = getUint16(this.buffer, recordOffset, le)
+		const recordSubType = getUint16(this.buffer, recordOffset + 2, le)
+
+        let recordLE = le
+		if (recordType === 1) {
+			if (recordSubType < 1 || recordSubType > 2) {
+				return
+			}
+			recordLE = (recordSubType === 2)
+		}
+
+		let recordContentOffset = getUint32(this.buffer, recordOffset + 12, le)
+		recordContentOffset += this.flirOffset
+
+		this.parseFlirFFFRecordTags(flir, recordType, recordContentOffset, recordLE)
+	}
+
+	parseFlirFFFRecordTags(flir, recordType, recordContentOffset, le) {
+	    if (!(recordType in tags.flir)) {
+			return
+		}
+
+		const recordTagsInfo = tags.flir[recordType];
+
+	    // TODO: What the fuck?!!
```

Comment:
No idea where this comes from, but it works consistently, so I'm going to leave it for now. Once we have more photos I could investigate in more detail maybe.